test/e2e/framework/kubectl/kubectl_builder.go

```diff
@@ -0,0 +1,352 @@
+func RunKubectlOrDieInput(namespace string, data string, args ...string) string {
+	tk := NewTestKubeconfig(framework.TestContext.CertDir, framework.TestContext.Host, framework.TestContext.KubeConfig, framework.TestContext.KubeContext, framework.TestContext.KubectlPath, namespace)
+	return NewKubectlCommand(tk, args...).WithStdinData(data).ExecOrDie(namespace)
+}
+
+// RunKubectlInput is a convenience wrapper over Builder that takes input to stdin
+func RunKubectlInput(namespace string, data string, args ...string) (string, error) {
+	tk := NewTestKubeconfig(framework.TestContext.CertDir, framework.TestContext.Host, framework.TestContext.KubeConfig, framework.TestContext.KubeContext, framework.TestContext.KubectlPath, namespace)
+	return NewKubectlCommand(tk, args...).WithStdinData(data).Exec()
+}
+
+// RunKubemciWithKubeconfig is a convenience wrapper over RunKubemciCmd
+func RunKubemciWithKubeconfig(kubeconfig string, projectID string, args ...string) (string, error) {
+	if kubeconfig != "" {
+		args = append(args, "--"+clientcmd.RecommendedConfigPathFlag+"="+kubeconfig)
+	}
+	return RunKubemciCmd(projectID, args...)
+}
+
+// RunKubemciCmd is a convenience wrapper over Builder to run kubemci.
+// It assumes that kubemci exists in PATH.
+func RunKubemciCmd(projectID string, args ...string) (string, error) {
+	// kubemci is assumed to be in PATH.
+	kubemci := "kubemci"
+	b := new(Builder)
+	args = append(args, "--gcp-project="+projectID)
+
+	b.cmd = exec.Command(kubemci, args...)
+	return b.Exec()
+}
+
+// CreateEmptyFileOnPod creates empty file at given path on the pod.
+// TODO(alejandrox1): move to subpkg pod once kubectl methods have been refactored.
+func CreateEmptyFileOnPod(namespace string, podName string, filePath string) error {
+	_, err := RunKubectl(namespace, "exec", fmt.Sprintf("--namespace=%s", namespace), podName, "--", "/bin/sh", "-c", fmt.Sprintf("touch %s", filePath))
+	return err
+}
+
+// DumpDebugInfo dumps debug info of tests.
+func DumpDebugInfo(c clientset.Interface, ns string) {
+	sl, _ := c.CoreV1().Pods(ns).List(context.TODO(), metav1.ListOptions{LabelSelector: labels.Everything().String()})
+	for _, s := range sl.Items {
+		desc, _ := RunKubectl(ns, "describe", "po", s.Name, fmt.Sprintf("--namespace=%v", ns))
+		framework.Logf("\nOutput of kubectl describe %v:\n%v", s.Name, desc)
+
+		l, _ := RunKubectl(ns, "logs", s.Name, fmt.Sprintf("--namespace=%v", ns), "--tail=100")
+		framework.Logf("\nLast 100 log lines of %v:\n%v", s.Name, l)
+	}
+}
+
+// lookForString looks for the given string in the output of fn, repeatedly calling fn until
+// the timeout is reached or the string is found. Returns last log and possibly
+// error if the string was not found.
+// TODO(alejandrox1): move to pod/ subpkg once kubectl methods are refactored.
+func lookForString(expectedString string, timeout time.Duration, fn func() string) (result string, err error) {
+	for t := time.Now(); time.Since(t) < timeout; time.Sleep(framework.Poll) {
```

Comment:
let's define `framework.Poll`  within the package instead of importing it - reduce the amount of code e2ekubectl depends on the framework for.